src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -70,6 +80,31 @@ internal void PopulateAutoCompleteCandidates()
             }
         }
 
+        internal void SearchAutoCompleteCandidates(string input)
+        {
+            if (PortViewModel == null) return;
+
+            FilteredResults = searchElementsCache.AsEnumerable().Where(e =>
+                                QuerySearchElements(e, input)).Select(e =>
+                                {
+                                      var vm = new NodeSearchElementViewModel(e, this);
+                                      vm.RequestBitmapSource += SearchViewModelRequestBitmapSource;
+                                      return vm;
+                                });
+        }
```

Comment:
I think you can return search elements here and make another function including the part to convert searchelements to FilteredResults so we have less duplicated code, what do you think?

Reply:
I guess I mean if there is a way to convert the duplicate code her (with line 76) to a shared function

Reply:
Yes changing this to a shared function.